thorlcr/activities/keyedjoin/thkeyedjoinslave.cpp

```diff
@@ -1572,8 +1572,13 @@ class CKeyedJoinSlave : public CSlaveActivity, implements IJoinProcessor, implem
                     crc = 0;
                 RemoteFilename rfn;
                 part.getFilename(copy, rfn);
+
+                // NB: normally, a remote handler would have the part locally, so only the local path (getLocalPath) is needed.
+                // However, if the files are off-cluster and stripedOutOfClusterLookups is set, then the full remote path is needed.
+                // So we send the full remote path in all cases, but if local it will devolve to local path at the other end.
                 StringBuffer fname;
-                rfn.getLocalPath(fname);
+                rfn.getRemotePath(fname);
```

Comment:
Not directly related to this, but what will happen to filenames that use azureblob:plane// ?  I think isUrl will return false so it will have a ip prefixed to it.

Reply:
you're right. Irregular files like plugin files, will be messed up by this.
I'm going to change it so it tests if local or not here, and calls getLocalPath or getRemotePath accordingly.